### PR TITLE
Remove `doc/help` directory in the source tree after installation

### PR DESCRIPTION
**Pull request by steffengraber**
Fix for #574.

### CODE REVIEW

File: extras/help_generator/parse_help.py [hunk 1]
```diff
@@ -135,3 +135,4 @@
 
 shutil.rmtree(install_dir, ignore_errors=True)
 shutil.copytree(helpdir, install_dir)
+shutil.rmtree(helpdir, ignore_errors=True)
```
Comment (heplesser) [inline extras/help_generator/parse_help.py]: Wouldn't it be more efficient to use `shutil.move()` instead of `copytree()` followed by `rmtree()`?

Comment (steffengraber) [inline extras/help_generator/parse_help.py]: Yes, better. I changed it.

**Comment by jougs** [comment]
I think the actual problem with #574 is that help is generated during the installation step (which may be run by root) rather than earlier on (where things are normally run by an unprivileged user). In this sense, this PR just addresses a symptom and not the cause of the problem.

If help was generated after compilation, the resulting files would belong to the user which ran `make all` and installation could then be just a plain copy as it is already for most other artifacts. That final step could be run by root or normal user without any unclear consequences for file ownerships in the build directory.

@steffengraber: do you think you can make this change? This would involve changing the [current hook for help extraction](https://github.com/steffengraber/nest-simulator/blob/eb34551fabdb8f5da2150652ff2ed53d12963d33/cmake/GenerateHelp.cmake#L51) to a [build event](https://cmake.org/cmake/help/v3.2/command/add_custom_command.html#build-events) added to the appropriate target and adding the generated help to an appropriate [install](https://cmake.org/cmake/help/v3.2/command/install.html?highlight=install#installing-directories) block in the `doc/CMakeLists.txt` file. Something similar would also be required for the build system of the `MyModule` example.

If you have troubles with this, I can implement the changes. It is probably anyway my fault that this problem exists at all ;-)

**Comment by steffengraber** [comment]
@jougs Sorry, I know nothing about cmake. It would be nice if you could do the changes.

**Comment by jougs** [comment]
Replaced by #647.

**Pull request closed without merging**